subiquity/cloudinit.py

```diff
@@ -67,6 +69,56 @@ def read_legacy_status(stream):
     return None
 
 
+def read_recoverable_errors(
+    stream: bytes | str,
+    level: Optional[str] = None,
+) -> list[str]:
+    """Retrieve recoverable errors from cloud-init status."""
+
+    try:
+        status: dict[str, Any] = json.loads(stream)
+    except json.JSONDecodeError:
+        return []
+
+    errors_by_level: dict[str, list[str]] = status.get("recoverable_errors", {})
+
+    if level is not None:
+        return errors_by_level.get(level, [])
+
+    all_errors: list[str] = []
+    for _level in errors_by_level:
+        all_errors.extend(errors_by_level[_level])
+
+    return all_errors
+
+
+async def get_schema_failure_sources() -> list[str]:
+    """Retrieve the keys causing schema failure."""
+
+    cmd: list[str] = ["cloud-init", "schema", "--system"]
+    status_coro: Awaitable = arun_command(cmd)
+    try:
+        sp: CompletedProcess = await asyncio.wait_for(status_coro, 600)
```

Comment:
So while the big wait on `cloud-init status --wait` is allowed to be 10 minutes, most of the others should be instantaneous.  Can you do some real vm testing on the typical response times for these calls?  I'm thinking `timeout=10` might be more realistic.  Part of that 10 minutes is the expectation that the system may still be starting, cloud-init is fetching the data from the datasource, etc.  Whereas the calls here should be much quicker.

Reply:
Good point. In my tests they seem pretty instantaneous, even with a large config with many incorrect keys. I've updated the timeouts to the suggested 10 seconds.